src/main/java/org/commonwl/view/researchobject/ROBundleService.java

```diff
@@ -150,13 +152,19 @@ public Bundle createBundle(Workflow workflow, GitDetails gitInfo) throws IOExcep
             Set<HashableAgent> authors = new HashSet<>();
 
             boolean safeToAccess = gitSemaphore.acquire(gitInfo.getRepoUrl());
+            Git gitRepo = null;
             try {
-                Git gitRepo = gitService.getRepository(workflow.getRetrievedFrom(), safeToAccess);
+                gitRepo = gitService.getRepository(workflow.getRetrievedFrom(), safeToAccess);
                 Path relativePath = Paths.get(FilenameUtils.getPath(gitInfo.getPath()));
                 Path gitPath = gitRepo.getRepository().getWorkTree().toPath().resolve(relativePath);
                 addFilesToBundle(gitInfo, bundle, bundlePath, gitRepo, gitPath, authors, workflow);
+            } catch (GitAPIException | IOException e) {
+                org.commonwl.view.util.FileUtils.deleteBundleTemporaryDirectory(bundle);
+                org.commonwl.view.util.FileUtils.deleteBundleParentDirectory(bundle);
+                throw e;
             } finally {
                 gitSemaphore.release(gitInfo.getRepoUrl());
+                org.commonwl.view.util.FileUtils.deleteGitRepository(gitRepo);
```

Comment:
Remove git folder in the `finally` statement since we won't need it after this point (i.e. the graphs are created, the cwltool has been executed and populated the DB, and now the git artefacts are already in the bundle).